app/src/main/java/com/scv/slackgo/activities/LocationActivity.java

```diff
@@ -380,7 +388,7 @@ private void editLocation() {
 
     private boolean isValidLocation(Location location) {
         boolean isValid = true;
-        if (location.getName().isEmpty() || (location.getChannels().size() == 0) || (location.getChannels() == null)) {
+        if (location.getName().isEmpty() || (location.getChannelsByName().size() == 0) || (location.getChannelsByName() == null)) {
```

Comment:
We can instantiate channelsByName on Location's constructor, to avoid the null check.